Dcumentation/PixelmonEvolution/README.md

```diff
@@ -0,0 +1,213 @@
+* Call the `mintEvolvedPixelmon` method in the Pixelmon NFT contract and mint a pixelmon token for the caller of the Evolve method
+
+**Note** 
+* The evolve does not take input as batch. We can not provide multiple token ID in an array for batch evolve method.
+
+## Product document
+* [Evolution 2 Pixelmon](https://docs.google.com/document/d/1VsyHtsgHSxEljc-UfKdUyCZwY4ToL-EK5FSodybV6xI/edit#heading=h.gjdgxs)
+* [Evolution with hard staking](https://docs.google.com/document/d/1VsyHtsgHSxEljc-UfKdUyCZwY4ToL-EK5FSodybV6xI/edit?usp=sharing)
+
+# Proposed solution
+To Resolve the issue of batch evolve pixelmon NFT. We are creating the new Pixelmon Evolution Smart Contract. 
+
+**The Functionality of Pixelmon Evolution Smart Contract**
+1. It will have a `evolvePixelmon` method. This method will take the Pixelmon and Serum Token information in batch.
+2. It will be able to validate the pixelmon, serum token, token owner information, nonce, staking time period with the signature.
+3. Then it will transfer the serum token to the burner address `0x000000000000000000000000000000000000dEaD`.
+4. The it will call the `mintEvolvedPixelmon` method in the pixelmon NFT smart contract.
+5. the new evolved pixelmon will be minted and transferred to the `PixelmonEvolution` (this new) smart contract. And it's information (by whom and when it can be claimed) will be saved in the smart contract.
+6. The Pixelmon NFT that has been used for evolution will also be staked in this `PixelmonEvolution` (this new) smart contract for 30 day time period.
+7. User will be able to claim the E1 and E2 Pixelmon after the staking time lock period.
+8. We will keep a time lock flag. if the flag is not active then user will be able to claim it any time. This flag will be updatable by the contract owner. 
+5. The contract owner will be able to set the signer wallet address.
+6. The contract will be able to set the pixelmon contract address and the serum token address.
+
+
+### Usecase Diagram
+![Use Case Diagram](./Images/Use%20case%20diagram.drawio2.png "Use Case Diagram")
+
+### Activity Diagram for `evolvePixelmon` Method
+![Activity Diagram](./Images/Activity%20Diagramdrawio.drawio2.png "Activity Diagram")
+
+### Sequence Diagram
+![Sequence Diagram](./Images/Sequese%20Diagram.drawio2.png "Sequence Diagram")
+
+# Test plan
+## Test Cases
+```
+  Pixelmon Evolution Contract Functionality
+    ✔ Contract should be deployed 
+    ✔ recoverSignerFromSignature will recover signer wallet from signature
+    ✔ recoverSignerFromSignature will not recover signer wallet from signature for invalid data
+    ✔ Only contract owner will be able to setPixelmonAddress
+    ✔ Only contract owner will be able to setNextEvolvePixelmonId 
+    ✔ Only contract owner will be able to setSerumAddress
+    ✔ Only contract owner will be able to setSignerAddress
+    ✔ evolvePixelmon will revert if serumIds and amounts length is not same 
```

Comment:
added duplicate serum / pixelmon ID.

But in a normal scenario, it should not happen. because the backed will have the check and then they will create the signature